assessments/projects/custodian/joint-review.md

```diff
@@ -0,0 +1,900 @@
+| Object Storage      | “Object Storage is storage for the Internet. It is designed to make web-scale computing easier for developers.  Cloud Storage has a simple web services interface that you can use to store and retrieve any amount of data, at any time, from anywhere on the web. It gives any developer access to the same highly scalable, reliable, fast, inexpensive data storage infrastructure that Amazon uses to run its own global network of websites. The service aims to maximize benefits of scale and to pass those benefits on to developers.” |
+| Cloud Functions     | “Serverless Computing is a compute service that lets you run code without provisioning or managing servers. Functions execute your code only when needed and scales automatically, from a few requests per day to thousands per second. You pay only for the compute time you consume - there is no charge when your code is not running. With AWS Lambda, you can run code for virtually any type of application or backend service - all with zero administration.”|
+| Configuration Rules | “Using Configurations, you can assess your resource configurations and resource changes for compliance against the built-in or custom rules.Create your own custom rules in functions as a service that define your internal best practices and guidelines for resource configurations.”|
+
+#### Security Functions and Features
+
+The project makes the following assumptions regarding mature operational practices when using Custodian in order to limit the attack surface:
+
+* Hardened compute execution environment with minimal additional software
+  components running and minimal inbound access.
+* Custodian requires cloud credentials (role, service account/principal) with
+  elevated permissions for introspection and remediation. Access to those
+  credentials should be restricted to the Custodian compute execution
+  environment.
+* Custodian cloud credentials should be limited in scope to what is necessary
+  for custodian policy execution, i.e. minimized access.
+* As the policy files are considered trusted inputs -- and per IaaC best
+  practices -- an audit trail is implemented to validate the assurance of the
+  supply chain through rigorous version and change control mechanisms.
+
+At a project level, to mitigate primary threats to the project and code, the
+project uses GitHub’s security alerts (dependabot) scanning for CVEs on the
+dependency graph. Additionally, PRs are reviewed by a growing number of
+committers to prevent malicious code to be accepted into the code base.
+
+Beyond these, there are few security-specific features, as the project assumes
+the host cloud and corporate environment itself is hardened and trusted.  It
+remains to see how useful these assumptions would be in a Zero Trust
+Architecture (ZTA).
```

Comment:
Is there a way we can balance this?  Earlier on we discuss how c7n was designed with security in mind, but here we don't list ANY of the security design elements that showcase that.  Instead we make it sound like it is insecure and relies entirely on the environment to provide security of the project.
Ideally this would list pivot or leverage point that the c7n team should pay attention to when reviewing PRs as they could negatively (or positively) impact how secure c7n is.

Reply:
I think this is a real tension in the project vs the TAG goals - c7n consistently and transparently (to its credit) bills itself as a security tool that assumes (and requires) that the user has deep, expert level security knowledge to use the tool correctly.  it's not for novices.  the TAG goal is to make it easy/easier for novice users.  so these are actually out of balance.

Reply:
Your explanation makes more sense to me.  Is there a way we can call that out more, this is a security tool that does requires _increased_ diligence from the user to understand the tools configuration and capabilities throughly?